### PR TITLE
[IMP] hr_expense: UX improvements

### PR DESCRIPTION
**Pull request by sofiagvaladze**

### CODE REVIEW

File: addons/hr_expense/static/src/scss/hr_expense.scss [hunk 1]
```diff
@@ -18,3 +18,19 @@
...
+    z-index: 2;
+    left: 0;
+    top: 0;
+    i {
+      justify-content: center;
+      display: flex;
+      align-items: center;
+      height: 100%;
+    }
+}
```
Comment (phwa-odoo) [inline addons/hr_expense/static/src/scss/hr_expense.scss:36]: New line missing at the end of the file. These can be avoided by enabling the following setting in VSCode
image

File: addons/hr_expense/static/src/views/list.xml [hunk 2]
```diff
@@ -0,0 +1,57 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<templates xml:space="preserve">
+    <t t-name="hr_expense.ListButtons" t-inherit="web.ListView.Buttons" t-inherit-mode="primary" owl="1">
+
+       <!-- hr.expense and hr.expense.sheet -->
```
Comment (phwa-odoo) [inline addons/hr_expense/static/src/views/list.xml:5]: Not aligned

File: addons/hr_expense/static/src/views/list.xml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        <xpath expr="//button[hasclass('o_list_button_add')]" position="after">
+            <button type="button" class="d-none d-md-inline o_button_upload_expense btn btn-primary mx-1" t-on-click.prevent="uploadDocument">
+                Scan
+            </button>
```
Comment (phwa-odoo) [inline addons/hr_expense/static/src/views/list.xml:9]: Trailing whitespace

File: addons/hr_expense/static/src/views/list.xml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+            <input type="file" name="ufile" class="d-none" t-ref="fileInput" multiple="1" accept="*" t-on-change="onChangeFileInput"/>
+        </xpath>
+
+        <!-- hr.expense -->
+        <xpath expr="//button[hasclass('o_button_upload_expense')]" position="after">
+            <button t-if="displayCreateReport()" class="d-none d-md-block btn btn-secondary" t-on-click="onCreateReportClick">
+                Create Report
+            </button>
+        </xpath>
+
```
Comment (phwa-odoo) [inline addons/hr_expense/static/src/views/list.xml:22]: Trailing whitespaces

File: addons/hr_expense/static/src/views/list.xml [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        <!-- hr.expense -->
+        <xpath expr="//button[hasclass('o_button_upload_expense')]" position="after">
+            <button t-if="displayCreateReport()" class="d-none d-md-block btn btn-secondary" t-on-click="onCreateReportClick">
+                Create Report
+            </button>
+        </xpath>
+        
+         <!-- hr.expense.sheet -->
```
Comment (phwa-odoo) [inline addons/hr_expense/static/src/views/list.xml:23]: Not aligned

**Review by phwa-odoo — COMMENTED** [review]
Minor cosmetic comments :innocent: Don't forget to put the PR in ready to review

File: addons/hr_expense/models/hr_expense.py [hunk 6]
```diff
@@ -479,14 +479,27 @@ def _get_default_expense_sheet_values(self):
...
+        if self:
+            expenses = self.filtered(lambda e: e.state == 'draft' and not e.sheet_id)
+        else:
+            expenses = self.env['hr.expense'].search([('state', '=', 'draft'), ('sheet_id', '=', False), ('employee_id', '=', self.env.user.employee_id.id)])
+
+        if not expenses:
+            raise UserError(_('You have no expense to report'))
+        else:
+            result = expenses.action_submit_expenses()
+            return result
```
Comment (kba-odoo) [inline addons/hr_expense/models/hr_expense.py:493]:
```suggestion
        return expenses.action_submit_expenses()
```

File: addons/hr_expense/models/hr_expense.py [hunk 7]
```diff
@@ -848,6 +861,7 @@ def _default_bank_journal_id(self):
 
     name = fields.Char('Expense Report Summary', required=True, tracking=True)
     expense_line_ids = fields.One2many('hr.expense', 'sheet_id', string='Expense Lines', copy=False)
+    product_ids = fields.Many2many('product.product', compute='_compute_product_ids', store=True, string='Categories', domain="[('can_be_expensed', '=', True), '|', ('company_id', '=', False), ('company_id', '=', company_id)]")
```
Comment (kba-odoo) [inline addons/hr_expense/models/hr_expense.py:864]: The domain is not needed as the field is not editable :innocent: 

```suggestion
    product_ids = fields.Many2many('product.product', compute='_compute_product_ids', store=True, string='Categories')
```

File: addons/sale_expense/models/product_template.py [hunk 8]
```diff
@@ -1,12 +1,32 @@
...
 
-from odoo import api, models
+from odoo import api, models, fields, _
 
 
 class ProductTemplate(models.Model):
     _inherit = 'product.template'
 
+    expense_policy_tooltip = fields.Char(compute='_compute_expense_policy_tooltip')
+
```
Comment (kba-odoo) [inline addons/sale_expense/models/product_template.py:11]:
```suggestion
    @api.depends_context('lang')
```

File: addons/sale_expense/views/product_view.xml [hunk 9]
```diff
@@ -8,8 +8,11 @@
         <field name="arch" type="xml">
             <xpath expr="//group[@name='product_details']" position="inside">
                 <group string="Invoicing">
-                    <field name="invoice_policy" widget="radio"/>
                     <field name="expense_policy" widget="radio"/>
+                    <label for="expense_policy_tooltip" string=""/>
+                    <div class="o_row">
+                        <field name ="expense_policy_tooltip" class="fst-italic text-muted"/>
```
Comment (kba-odoo) [inline addons/sale_expense/views/product_view.xml:14]:
```suggestion
                        <field name="expense_policy_tooltip" class="fst-italic text-muted"/>
```

File: addons/hr_expense/models/hr_expense.py [hunk 10]
```diff
@@ -841,6 +853,7 @@ def _default_bank_journal_id(self):
 
     name = fields.Char('Expense Report Summary', required=True, tracking=True)
     expense_line_ids = fields.One2many('hr.expense', 'sheet_id', string='Expense Lines', copy=False)
+    product_ids = fields.Many2many('product.product', compute='_compute_product_ids', store=True, string='Categories')
```
Comment (KangOl) [inline addons/hr_expense/models/hr_expense.py:856]: Does it need to be stored?

Comment (sofiagvaladze) [reply addons/hr_expense/models/hr_expense.py]: Indeed, no need to have it stored

**Comment by kba-odoo** [comment]
robodoo r+

**Pull request closed without merging**